crates/client/data-availability/src/ethereum/config.rs

```diff
@@ -1,68 +1,28 @@
 use std::fs::File;
 use std::path::PathBuf;
 
+use mc_eth_client::config::{EthereumProviderConfig, EthereumWalletConfig, StarknetContracts};
 use serde::{Deserialize, Serialize};
 
 use crate::{DaError, DaMode};
 
-pub const DEFAULT_ETHEREUM_NODE: &str = "127.0.0.1:8545";
-// default key derived from starting anvil as follows:
-// anvil -b 5 --config-out $BUILD_DIR/anvil.json
-// PRE_PRIVATE=$(jq -r '.private_keys[0]' $BUILD_DIR/anvil.json)
-pub const DEFAULT_SEQUENCER_KEY: &str = "ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
-pub const DEFAULT_STARKNET_CORE_CONTRACTS: &str = "0x5FbDB2315678afecb367f032d93F642f64180aa3";
-pub const DEFAULT_CHAIN_ID: u64 = 31337;
-
-#[derive(Clone, PartialEq, Serialize, Deserialize, Debug)]
-pub struct EthereumConfig {
-    #[serde(default = "default_http")]
-    pub http_provider: String,
-    #[serde(default = "default_core_contracts")]
-    pub core_contracts: String,
-    #[serde(default = "default_sequencer_key")]
-    pub sequencer_key: String,
-    #[serde(default = "default_chain_id")]
-    pub chain_id: u64,
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct EthereumDaConfig {
+    #[serde(default)]
+    pub provider: EthereumProviderConfig,
+    #[serde(default)]
+    pub wallet: Option<EthereumWalletConfig>,
+    #[serde(default)]
+    pub contracts: StarknetContracts,
     #[serde(default)]
     pub mode: DaMode,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub poll_interval_ms: Option<u64>,
 }
 
-impl TryFrom<&PathBuf> for EthereumConfig {
+impl TryFrom<&PathBuf> for EthereumDaConfig {
```

Comment:
Let's not use TryFrom for reading data from disk.
Let's keep it for in memory data type conversion.